src/main/java/org/ibp/api/java/impl/middleware/germplasm/validator/GermplasmNameValidator.java

```diff
@@ -0,0 +1,147 @@
+package org.ibp.api.java.impl.middleware.germplasm.validator;
+
+import org.apache.commons.lang.StringUtils;
+import org.generationcp.middleware.api.germplasm.GermplasmNameService;
+import org.generationcp.middleware.domain.germplasm.GermplasmNameRequestDto;
+import org.generationcp.middleware.pojos.Name;
+import org.generationcp.middleware.pojos.UserDefinedField;
+import org.generationcp.middleware.util.Util;
+import org.ibp.api.exception.ApiRequestValidationException;
+import org.ibp.api.java.impl.middleware.common.validator.GermplasmValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.MapBindingResult;
+
+import java.text.ParseException;
+import java.util.HashMap;
+
+@Component
+public class GermplasmNameValidator {
```

Comment:
It should be named GermplasmNameRequestValidator

Reply:
done.